### PR TITLE
Created Azure.Messaging.ServiceBus samples to be used by tutorials in DOCS

### PR DESCRIPTION
**Pull request by spelluru**
## Description
Created Track 2 version of samples referred by these two tutorials in DOCS:

https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-tutorial-topics-subscriptions-portal
https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-to-event-grid-integration-example?tabs=v2

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [X] Title of the pull request is clear and informative.
- [X] If applicable, the code is properly documented.
- [ X] The code builds without any errors.

Here's the direct link to the folder I created. I created a folder named Azure.Messaging.ServiceBus under DotNet.  
https://github.com/spelluru/azure-service-bus/tree/sbustrack2sdk0819/samples/DotNet/Azure.Messaging.ServiceBus

### CODE REVIEW

**Comment by spelluru** [comment]
@axisc

File: samples/DotNet/Azure.Messaging.ServiceBus/BasicSendReceiveTutorialwithFilters/BasicSendReceiveTutorialWithFilters/Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using Azure.Messaging.ServiceBus;
+using System;
+
+namespace BasicSendReceiveTutorialWithFilters
+{
+    public static class Extensions
+    {
+        public static T As<T>(this ServiceBusReceivedMessage message) where T : class
+        {
+            return JsonConvert.DeserializeObject<T>(Encoding.UTF8.GetString(message.Body.ToBytes().ToArray()));
```
Comment (SeanFeldman) [inline samples/DotNet/Azure.Messaging.ServiceBus/BasicSendReceiveTutorialwithFilters/BasicSendReceiveTutorialWithFilters/Extensions.cs:15]: Any reason why [`System.Text.Json`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json) is not used?

File: samples/DotNet/Azure.Messaging.ServiceBus/BasicSendReceiveTutorialwithFilters/BasicSendReceiveTutorialWithFilters/SendReceive.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        public string[] Store;
+        public int NrOfMessagesPerStore;
+
+        public async Task SendMessages()
+        {
+            try
+            {
+
+                await using (ServiceBusClient client = new ServiceBusClient(ServiceBusConnectionString))
+                {
```
Comment (SeanFeldman) [inline samples/DotNet/Azure.Messaging.ServiceBus/BasicSendReceiveTutorialwithFilters/BasicSendReceiveTutorialWithFilters/SendReceive.cs:28]: Is this scope needed? Could be done with `await using var client = new ServiceBusClient(...);`

File: samples/DotNet/Azure.Messaging.ServiceBus/BasicSendReceiveTutorialwithFilters/BasicSendReceiveTutorialWithFilters/SendReceive.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            {
+                taskList.Add(this.ReceiveMessages(Subscriptions[i]));
+            }
+
+            await Task.WhenAll(taskList);
+        }
+
+        private async Task ReceiveMessages(string subscription)
+        {
+            await using (ServiceBusClient client = new ServiceBusClient(ServiceBusConnectionString))
```
Comment (SeanFeldman) [inline samples/DotNet/Azure.Messaging.ServiceBus/BasicSendReceiveTutorialwithFilters/BasicSendReceiveTutorialWithFilters/SendReceive.cs:82]: Same as above

File: samples/DotNet/Azure.Messaging.ServiceBus/MessageSender/Program.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
+﻿using Azure.Messaging.ServiceBus;
+using System;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MessageSender
+{
+    class Program
+    {
+        const string ServiceBusConnectionString = "Endpoint=sb://spsbusegridtutns.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=j1lrBlYD/6QXwC+OqYpByxoq6WZQ7IgMkV37ETlTUUc=";
```
Comment (SeanFeldman) [inline samples/DotNet/Azure.Messaging.ServiceBus/MessageSender/Program.cs:10]: Ooops

File: samples/DotNet/Azure.Messaging.ServiceBus/MessageSender/Program.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            await SendMessagesAsync(numberOfMessages);
+
+            Console.ReadKey();
+        }               
+
+        static async Task SendMessagesAsync(int numberOfMessagesToSend)
+        {
+            try
+            {
+                await using (ServiceBusClient client = new ServiceBusClient(ServiceBusConnectionString))
```
Comment (SeanFeldman) [inline samples/DotNet/Azure.Messaging.ServiceBus/MessageSender/Program.cs:36]: Same as above

File: samples/DotNet/Azure.Messaging.ServiceBus/ServiceBusEventGridIntegrationV2/FunctionApp1/ReceiveMessagesOnEvent.cs [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+using Azure.Messaging.ServiceBus;
+using Microsoft.Azure.WebJobs.Extensions.EventGrid;
+using Microsoft.Azure.EventGrid.Models;
+
+namespace SBEventGridIntegrationV2
+{
+    public static class ReceiveMessagesOnEvent
+    {
+        const string ServiceBusConnectionString = "Endpoint=sb://spsbusegridtutns.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=j1lrBlYD/6QXwC+OqYpByxoq6WZQ7IgMkV37ETlTUUc=";
```
Comment (SeanFeldman) [inline samples/DotNet/Azure.Messaging.ServiceBus/ServiceBusEventGridIntegrationV2/FunctionApp1/ReceiveMessagesOnEvent.cs:26]: Reset the key?

**Review by SeanFeldman — CHANGES_REQUESTED** [review]

**Comment by spelluru** [comment]
@SeanFeldman - Thanks for the feedback. Reset the key today :). I will fix other issues. I am trying to convert the Azure Function to v3.0 and use only Event Grid Trigger instead of HTTP trigger with the function. I will upload changes in a day or two.

**Comment by spelluru** [comment]
@SeanFeldman - I have fixed the issues except the one about System.Text. I will look into it in another PR. Thanks.

**Pull request merged**